src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java

```diff
@@ -59,4 +78,14 @@ public Submission markSubmissionStatus(String submissionId, SubmissionStatus sta
 
         return submissionRepository.save(submission);
     }
+
+    public boolean checkUserHasAccessToSubmission(SubmissionUser user, String submissionId){
+        Optional<Submission> optSubmission = submissionRepository.findById(submissionId);
+        if(optSubmission.isPresent()){
+            SubmissionUser submissionUser = optSubmission.get().getUser();
+            return submissionUser.getId() == user.getId();
+        }else{
+            throw new RuntimeException("Given submission with id " + submissionId + "does not exist");
```

Comment:
```suggestion
            throw new RuntimeException("Given submission with id " + submissionId + " does not exist");
```

Also, I don't think this exception is being handled by the controller?

Reply:
handled